File: app/src/main/java/me/tagavari/airmessage/helper/MediaFileHelper.java
package me.tagavari.airmessage.helper;

import android.content.Context;
import android.media.MediaMetadataRetriever;
import android.net.Uri;

import java.io.File;

public class MediaFileHelper {
	/**
	 * Retrieves the duration of a media file (ie. audio)
	 * @param file the location of the file to check
	 * @return the duration of the media file in milliseconds
	 */
	public static long getMediaDuration(File file) {
		//Creating a new media metadata retriever
		MediaMetadataRetriever mmr = new MediaMetadataRetriever();
		
		try {
			//Setting the source file
			mmr.setDataSource(file.getPath());
			
			//Getting the duration
			return Long.parseLong(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION));
		} catch(RuntimeException exception) {
			//Printing the stack trace
			exception.printStackTrace();
		} finally {
			//Releasing the media metadata retriever
			mmr.release();
		}
		
		//Returning an invalid value
		return -1;
	}
	
	/**
	 * Retrieves the duration of a media file (ie. audio)
	 * @param context context to use to resolve the URI
	 * @param uri the uri of the file to check
	 * @return the duration of the media file in milliseconds
	 */
	public static long getMediaDuration(Context context, Uri uri) {
		//Creating a new media metadata retriever
		MediaMetadataRetriever mmr = new MediaMetadataRetriever();
		
		try {
			//Setting the source file
			mmr.setDataSource(context, uri);
			
			//Getting the duration
			return Long.parseLong(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION));
		} catch(RuntimeException exception) {
			//Printing the stack trace
			exception.printStackTrace();
		} finally {
			//Releasing the media metadata retriever
			mmr.release();
		}
		
		//Returning an invalid value
		return -1;
	}
}

File: common/me/tagavari/airmessage/common/helper/MediaFileHelper.kt
package me.tagavari.airmessage.common.helper

import android.content.Context
import android.media.MediaMetadataRetriever
import android.net.Uri
import java.io.File

object MediaFileHelper {
	/**
	 * Retrieves the duration of a media file (ie. audio)
	 * @param file the location of the file to check
	 * @return the duration of the media file in milliseconds
	 */
	@JvmStatic
	fun getMediaDuration(file: File): Long {
		//Creating a new media metadata retriever
		val mmr = MediaMetadataRetriever()
		try {
			//Setting the source file
			mmr.setDataSource(file.path)
			
			//Getting the duration
			return mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toLong() ?: -1L
		} catch(exception: RuntimeException) {
			//Printing the stack trace
			exception.printStackTrace()
			
			//Returning an invalid value
			return -1
		} finally {
			//Releasing the media metadata retriever
			mmr.release()
		}
	}
	
	/**
	 * Retrieves the duration of a media file (ie. audio)
	 * @param context context to use to resolve the URI
	 * @param uri the uri of the file to check
	 * @return the duration of the media file in milliseconds
	 */
	@JvmStatic
	fun getMediaDuration(context: Context, uri: Uri): Long {
		//Creating a new media metadata retriever
		val mmr = MediaMetadataRetriever()
		try {
			//Setting the source file
			mmr.setDataSource(context, uri)
			
			//Getting the duration
			return mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toLong() ?: -1L
		} catch(exception: RuntimeException) {
			//Printing the stack trace
			exception.printStackTrace()
			
			//Returning an invalid value
			return -1
		} finally {
			//Releasing the media metadata retriever
			mmr.release()
		}
	}
}